lib/frame.c

```diff
@@ -81,6 +81,9 @@ int quicly_decode_ack_frame(const uint8_t **src, const uint8_t *end, quicly_ack_
     frame->smallest_acknowledged = frame->largest_acknowledged - tmp;
     frame->ack_block_lengths[0] = tmp + 1;
 
+    if (frame->num_gaps > QUICLY_MAX_ACK_FRAME_RANGE)
+        frame->num_gaps = QUICLY_MAX_ACK_FRAME_RANGE;
+
```

Comment:
I do not think this works.

For the parsing of the ACK frame to succeed, we need to read ACK ranges specified by the frame.

It might be a good idea to add a unit test that uses ACK frames that exceed the max. t/frame.c has some tests that cover ACK frame decoding.

Reply:
Apologies. This wouldn't have happened if I better digested your original feedback. Yes, let's decode the entire frame, and only consume what we can into the buffer. Hopefully this is more in line with your thinking: c2c26eba205058dd6631fe6b5952626eecc53f44

Reply:
Hi @kazuho, I added a unit test for decoding a bogus synthetic ACK frame with number of ACK ranges exceeding the size of the internal buffer, in cfec91d1.